hail/python/hail/ir/ir.py

```diff
@@ -1359,20 +1359,104 @@ class Join(IR):
                       join_exprs=sequenceof(anytype),
                       join_func=func_spec(1, anytype))
     def __init__(self, virtual_ir, temp_vars, join_exprs, join_func):
-        super(Join, self).__init__(*(e._ir for e in join_exprs))
+        super(Join, self).__init__(virtual_ir, *(e._ir for e in join_exprs))
```

Comment:
so I don't think it's totally correct that this passes join_exprs as children.

We should write a pass in Python that executes Joins and replaces them with the proper IR. But that isn't necessary to make the change here.

Reply:
I think process_joins should do this -- take the IR as an argument as well, and return the IRs you should execute ( which don't have joins, we can make `render()` error out )